app/src/main/java/com/ucm/informatica/spread/Fragments/ProfileFragment.java

```diff
@@ -53,4 +133,113 @@ private void showSnackBar(String text) {
+            editAge.setText(ageText.getText());
+            editName.setVisibility(View.VISIBLE);
+            editAge.setVisibility(View.VISIBLE);
+        } else{
+            editProfileButton.setVisibility(View.VISIBLE);
+            saveProfileButton.setVisibility(View.GONE);
+            if(!editName.getText().toString().isEmpty())
+                nameText.setText(editName.getText());
+            if(!editAge.getText().toString().isEmpty())
+                ageText.setText(editAge.getText());
+            nameText.setVisibility(View.VISIBLE);
+            ageText.setVisibility(View.VISIBLE);
+            editName.setVisibility(View.GONE);
+            editAge.setVisibility(View.GONE);
+        }
+    }
+
+    public void saveData(){
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        if(!editName.getText().toString().isEmpty())
+            editor.putString("Name", editName.getText().toString());
+        if(!editAge.getText().toString().isEmpty())
+            editor.putString("Age", editAge.getText().toString());
+        editor.commit();
+    }
+
+    @Override
+    public void onRenderWatchwordView(Boolean edit) {
+        if (edit) {
+            editWatchwordButton.setVisibility(View.GONE);
+            saveWatchwordButton.setVisibility(View.VISIBLE);
+            watchwordMessageText.setVisibility(View.GONE);
+            watchwordResponesText.setVisibility(View.GONE);
+            editWatchwordMessage.setText(watchwordMessageText.getText());
+            editWatchwordResponse.setText(watchwordResponesText.getText());
+            editWatchwordMessage.setVisibility(View.VISIBLE);
+            editWatchwordResponse.setVisibility(View.VISIBLE);
+        } else{
+            editWatchwordButton.setVisibility(View.VISIBLE);
+            saveWatchwordButton.setVisibility(View.GONE);
+            if(!editWatchwordMessage.getText().toString().isEmpty())
+                watchwordMessageText.setText(editWatchwordMessage.getText());
+            if(!editWatchwordResponse.getText().toString().isEmpty())
+                watchwordResponesText.setText(editWatchwordResponse.getText());
+            watchwordMessageText.setVisibility(View.VISIBLE);
+            watchwordResponesText.setVisibility(View.VISIBLE);
+            editWatchwordMessage.setVisibility(View.GONE);
+            editWatchwordResponse.setVisibility(View.GONE);
+        }
+    }
+
+    public void saveWatchwordData(){
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        if(!editWatchwordMessage.getText().toString().isEmpty())
+            editor.putString("Message", editWatchwordMessage.getText().toString());
+        if(!editWatchwordResponse.getText().toString().isEmpty())
+            editor.putString("Response", editWatchwordResponse.getText().toString());
+        editor.commit();
+    }
+
+    public void loadData(){
+        nameText.setText(sharedPreferences.getString("Name", ""));
```

Comment:
"Age", "Shirt", "Pants" ... could be unified in a static class inside Constant file (which already [exists](https://github.com/patriciaTel/TFG/blob/master/app/src/main/java/com/ucm/informatica/spread/Utils/Constants.java) )